### PR TITLE
Fix branch dropdown shifting on page load

### PR DESCRIPTION
**Pull request by silverwind**
When HTML renders, whitspace within the button is significant and the browser will render it, but when Vue re-renders the button, it drops the whitespace, resulting in a layout shift on the button on load. Fix it by compressing the HTML and adding explicit whitespace where needed.

This unfortunately does make this code less readable, but it's a sacrifice I'm willing to make to avoid that layout shift. Ideally, Vue would not re-render the button in first place, but I'm not motivated to dig into that further.

### CODE REVIEW

**Comment by wxiaoguang** [comment]
https://pkg.go.dev/text/template#hdr-Text_and_spaces

`{{- ... -}}`

and

```
<tag>{{/*
*/}}<tag>
```

or

```
<tag><!--
--><tag>
```

**Comment by wxiaoguang** [comment]
"Ideally, Vue would not re-render the button in first place"

The first time you see the buttons, they are rendered by browser and original HTML, not by Vue.

**Comment by silverwind** [comment]
Ready again. Now using flexbox because inside a flexbox, inter-tag whitespace is insignificant.

Screen Shot 2022-10-17 at 22 59 42
Screen Shot 2022-10-17 at 22 59 51
Screen Shot 2022-10-17 at 23 00 06

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Maybe the `<span>&nbsp;</span>` could be replaced by a `margin-left` on the next element.

**Comment by silverwind** [comment]
Actually, I'm contemplating removing the label and do it like GitHub:

image

Anyone agree?

**Comment by wxiaoguang** [comment]
Agree, the icon is clear enough

**Comment by silverwind** [comment]
Ready for review again.

Removed the prefix and did some misc tweaks like changing to `<button>` for a11y and adding a bit of margin besides repo path:

image

image

File: templates/repo/branch_dropdown.tmpl [hunk 1]
```diff
@@ -14,19 +14,19 @@
 		data-branch-url-suffix="{{if .branchURLSuffix}}{{.branchURLSuffix}}{{else}}{{if $.root.TreePath}}/{{PathEscapeSegments $.root.TreePath}}{{end}}{{end}}"
 		data-tag-url-prefix="{{if .tagURLPrefix}}{{.tagURLPrefix}}{{else if $release}}{{$.root.RepoLink}}/compare/{{else}}{{$.root.RepoLink}}/{{if $.root.PageIsCommits}}commits{{else}}src{{end}}/tag/{{end}}"
 		data-tag-url-suffix="{{if .tagURLSuffix}}{{.tagURLSuffix}}{{else if $release}}...{{if $release.IsDraft}}{{PathEscapeSegments $release.Target}}{{else}}{{if $release.TagName}}{{PathEscapeSegments $release.TagName}}{{else}}{{PathEscapeSegments $release.Sha1}}{{end}}{{end}}{{else}}{{if $.root.TreePath}}/{{PathEscapeSegments $.root.TreePath}}{{end}}{{end}}">
-		<div class="branch-dropdown-button ellipsis ui basic small compact button" @click="menuVisible = !menuVisible" @keyup.enter="menuVisible = !menuVisible">
-			<span class="text">
+		<button class="branch-dropdown-button ellipsis ui basic small compact button df" @click="menuVisible = !menuVisible" @keyup.enter="menuVisible = !menuVisible">
```
Comment (delvh) [inline templates/repo/branch_dropdown.tmpl:17]: Ah great, even improved a11y!

**Review by delvh — APPROVED** [review]

**Comment by silverwind** [comment]
Margin between these 3 buttons is now also equal. I checked blame, it was related to render on release page, but no issues there with the margin removed.

image

**Review by lafriks — APPROVED** [review]

**Pull request merged**